### PR TITLE
update version block to display content

### PR DESCRIPTION
**Pull request by mirnawong1**
this PR fixes the version block so that the header AND content displays for v1.8 and higher. previously the header was displaying in the sidebar for all versions, but the actual content wasn't. this was a bit confusing and not the intended experience.

**Update**
version block was for 1.8 and left that in there. this pr now fixes the broken link in the migration guide.

### CODE REVIEW

**Comment by matthewshaver** [comment]
@mirnawong1 The version blocks were intentional. The ability to use both `test:` and `data_test:` is only available in v1.8. In v1.9 you won't be able to use `test:` anymore and we will have different messaging.

**Comment by mirnawong1** [comment]
> @mirnawong1 The version blocks were intentional. The ability to use both `test:` and `data_test:` is only available in v1.8. In v1.9 you won't be able to use `test:` anymore and we will have different messaging.

got it, thought it was the case based on [my update above](https://github.com/dbt-labs/docs.getdbt.com/pull/5154#issue-2205250926) and when the broken right toc is fixed, should sort it out. so i've removed the version change and keep the rest since the link for data test syntax is actually broken in the migration guide.

**Comment by mirnawong1** [comment]
hey @matthewshaver - can you rereview? thanks!

File: website/docs/docs/dbt-versions/core-upgrade/01-upgrading-to-v1.8.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ Unit tests are defined in YML files in your `models/` directory and are currentl
 
 #### New `data_tests:` syntax
 
-The `tests:` syntax is changing to reflect the addition of unit tests. Start migrating your [data test](/docs/build/data-tests#new-test-syntax) YML to use `data_tests:` after you upgrade to v1.8 to prevent issues in the future.
+The `tests:` syntax is changing to reflect the addition of unit tests. Start migrating your [data test](/docs/build/data-tests#new-data_tests-syntax) YML to use `data_tests:` after you upgrade to v1.8 to prevent issues in the future.
```
Comment (nghi-ly) [inline website/docs/docs/dbt-versions/core-upgrade/01-upgrading-to-v1.8.md:66]: this anchor link is still broken -- doesn't take me to the section directly

curious, i wonder if the code font in the section title is the root cause

**Review by nghi-ly — CHANGES_REQUESTED** [review]
this anchor link is still broken -- doesn't take me to the section directly

curious, i wonder if the code font in the section title is the root cause

**Comment by mirnawong1** [comment]
oh damn, i'll test it again. i wonder if it has something to do with the version of the doc. are you on v1.8? or maybe it has something to do with the colon 👀  will test!

**Comment by mirnawong1** [comment]
ok i've tried removing the colon, adding backets (e.g {new-data_tests-syntax}), changing it to 'new data tests syntax' without any characters or underscore, and removing the versions -- and nothing is working 🥲  i wonder if it has anything to do with the right toc but OR the fact that 1.8 is beta? idk

going to open an asana ticket for the frontend team: https://www.loom.com/share/a43cd0e46e984c6498a081e6cdfc7c9d?sid=f5d9fbd0-0617-406b-bd29-f3ca6d347749

**Comment by mirnawong1** [comment]
asana task here: https://app.asana.com/0/1200099998847559/1206949105408562/f

File: website/docs/docs/build/data-tests.md [hunk 2]
```diff
@@ -261,13 +261,21 @@ Note that, if you select to store test failures:
...
 - A test's results will always **replace** previous failures for the same test.
 
-<VersionBlock firstVersion="1.8" lastVersion="1.8">
+
 
 ## New `data_tests:` syntax
 
+<VersionBlock lastVersion="1.7">
+
+The `tests` config has been renamed to `data_tests` in dbt version 1.8. To read more about this change, switch to version v1.8 in the documentation navigation menu.
```
Comment (mirnawong1) [inline website/docs/docs/build/data-tests.md]:
```suggestion
In dbt version 1.8, we updated the `tests` configuration to `data_tests`. For detailed information on this update, please select version v1.8 from the documentation navigation menu.
```

File: website/docs/docs/build/data-tests.md [hunk 3]
```diff
@@ -261,13 +261,21 @@ Note that, if you select to store test failures:
...
 
+<VersionBlock lastVersion="1.7">
+
+In dbt version 1.8, we updated the `tests` configuration to `data_tests`. For detailed information on this update, please select version v1.8 from the documentation navigation menu.
+
+</VersionBlock>
+
+<VersionBlock firstVersion="1.8" lastVersion="1.8">
+  
 Data tests were historically called "tests" in dbt as the only form of testing available. With the introduction of unit tests in v1.8, it was necessary to update our naming conventions and syntax. As of v1.8,  `tests:` is still supported in your YML configuration files as an alias but will be deprecated in the future in favor of `data_tests:`.
```
Comment (mirnawong1) [inline website/docs/docs/build/data-tests.md]:
```suggestion
Data tests were historically called "tests" in dbt as the only form of testing available. With the introduction of unit tests in v1.8, it was necessary to update our naming conventions and syntax. 

As of v1.8, `tests:` is still supported in your YML configuration files as an alias but will be deprecated in the future in favor of `data_tests:`. 
```

File: website/docs/docs/build/data-tests.md [hunk 4]
```diff
@@ -261,13 +261,23 @@ Note that, if you select to store test failures:
...
 
-<VersionBlock firstVersion="1.8" lastVersion="1.8">
+
 
 ## New `data_tests:` syntax
 
-Data tests were historically called "tests" in dbt as the only form of testing available. With the introduction of unit tests in v1.8, it was necessary to update our naming conventions and syntax. As of v1.8,  `tests:` is still supported in your YML configuration files as an alias but will be deprecated in the future in favor of `data_tests:`. 
+<VersionBlock lastVersion="1.7">
+
+In dbt version 1.8, we updated the `tests` configuration to `data_tests`. For detailed information on this update, please select version v1.8 from the documentation navigation menu.
```
Comment (mirnawong1) [inline website/docs/docs/build/data-tests.md]:
```suggestion
In dbt version 1.8, we updated the `tests` configuration to `data_tests`. For detailed information, select version v1.8 from the documentation navigation menu.
```

File: website/docs/docs/build/data-tests.md [hunk 5]
```diff
@@ -125,7 +125,7 @@ You can find more information about these data tests, and additional configurati
 Those four tests are enough to get you started. You'll quickly find you want to use a wider variety of tests—a good thing! You can also install generic data tests from a package, or write your own, to use (and reuse) across your dbt project. Check out the [guide on custom generic tests](/best-practices/writing-custom-generic-tests) for more information.
 
 :::info
-There are generic tests defined in some open source packages, such as [dbt-utils](https://hub.getdbt.com/dbt-labs/dbt_utils/latest/) and [dbt-expectations](https://hub.getdbt.com/calogica/dbt_expectations/latest/) — skip ahead to the docs on [packages](/docs/build/packages) to learn more!
+There are generic tests defined in some open-source packages, such as [dbt-utils](https://hub.getdbt.com/dbt-labs/dbt_utils/latest/) and [dbt-expectations](https://hub.getdbt.com/calogica/dbt_expectations/latest/) — skip ahead to the docs on [packages](/docs/build/packages) to learn more!
```
Comment (matthewshaver) [inline website/docs/docs/build/data-tests.md]:
```suggestion
There are generic tests defined in some open-source packages, such as [dbt-utils](https://hub.getdbt.com/dbt-labs/dbt_utils/latest/) and [dbt-expectations](https://hub.getdbt.com/calogica/dbt_expectations/latest/) &mdash; skip ahead to the docs on [packages](/docs/build/packages) to learn more!
```

**Review by matthewshaver — APPROVED** [review]

**Review by nghi-ly — APPROVED** [review]

**Pull request merged**